src/smc_mallows_new_item_rank.cpp

```diff
@@ -163,7 +167,8 @@ Rcpp::List smc_mallows_new_item_rank(
     /* normalise weights ------------------------------------ */
     const double& maxw = max(log_inc_wgt);
     const vec& w = exp(log_inc_wgt - maxw);
-    const vec& norm_wgt = w / sum(w);
+    const vec& norm_wgt = w / sum(w);  
+    ESS_vec(tt) = 1/sum(norm_wgt^2);
```

Comment:
Can I assume this should also be `ESS_vec(tt) = (sum(norm_wgt) * sum(norm_wgt)) / sum(norm_wgt % norm_wgt);`?

Reply:
Actually, there's another problem here apart from the equation itself: This code chunk happens outside of a loop involving `tt`, so `ESS_vec(tt)` should not involve the iterator. A previous [comment from @osorensen](https://github.com/ocbe-uio/BayesMallows/pull/189#discussion_r846668455) suggested `ESS_vec(Time)` instead. Is that the case, @anjastein?

Reply:
> Can I assume this should also be `ESS_vec(tt) = (sum(norm_wgt) * sum(norm_wgt)) / sum(norm_wgt % norm_wgt);`?

Yes, it would be

Reply:
> Actually, there's another problem here apart from the equation itself: This code chunk happens outside of a loop involving `tt`, so `ESS_vec(tt)` should not involve the iterator. A previous [comment from @osorensen](https://github.com/ocbe-uio/BayesMallows/pull/189#discussion_r846668455) suggested `ESS_vec(Time)` instead. Is that the case, @anjastein?

I don't think this suggestion would store Time entries in the vector, but only one entry in the final position of the vector. I shall take a look now to see what can be done!

Reply:
> store Time entries in the vector

Ok, so if I understood correctly, what you want is the construction `rowvec(n_elem, fill_form)` ([source](http://arma.sourceforge.net/docs.html#Row)). I'm working on a commit implementing this, fingers crossed I can get it online within the next few minutes.

Reply:
Hi @anjastein, just implemented the aforementioned changes on commit 156ae3727b20690a25ccb320657659db78e368dd. Please check and confirm that it produces the desired behavior.

Reply:
Hi @wleoncio !
The `fill_form` can be `fill::zeros` because we will fill in each entry of `ESS_vec` in the `tt` loop with the calculation `ESS_vec(tt) = (sum(norm_wgt) * sum(norm_wgt)) / sum(norm_wgt % norm_wgt);` as `norm_wgt` gets recalculated in the `tt` loop as well. Otherwise, it looks good to me!

Reply:
OK, I've just pushed commit b46a3f5ab7664972d5de9b13386a4494d5af32d9 containing the changes mentioned above and [here](https://github.com/ocbe-uio/BayesMallows/commit/156ae3727b20690a25ccb320657659db78e368dd#r72662511). So `ESS_vec` starts with `ess_init` on index 0 and zero on indices {1, ... Time - 1}. Check?

Reply:
Looks good!